src/cdk-experimental/drag-drop/drag.ts

```diff
@@ -103,14 +103,15 @@ export class CdkDrag<T = any> implements OnDestroy {
   /** Element that will be used as a template to create the draggable item's preview. */
   @ContentChild(CdkDragPreview) _previewTemplate: CdkDragPreview;
 
-  /**
-   * Template for placeholder element rendered to show where a draggable would be dropped.
-   */
+  /** Template for placeholder element rendered to show where a draggable would be dropped. */
   @ContentChild(CdkDragPlaceholder) _placeholderTemplate: CdkDragPlaceholder;
 
   /** Arbitrary data to attach to this drag instance. */
   @Input() data: T;
 
+  /** Locks the position of the dragged element along the specified axis. */
+  @Input('cdkDragLockPosition') lockPosition: 'x' | 'y';
```

Comment:
What's your plan for API to support constraining to a specific region in addition to an axis? I wondering if it might make sense to have one API for both.

If it stays its own thing, though, how about `lockAxis` instead?

Reply:
I was thinking about having a separate input for constraining it to an area since you might want to enable both. I'm just not sure exactly what the type for the other input would be, because it would also need to capture constraining it to the viewport (`document` or `document.body`).

`lockAxis` sounds good as well.

Reply:
That works for me